README.md

```diff
@@ -112,3 +121,138 @@ make
+
+Tessel's ports can be used as module ports as in Tessel 1 (e.g. `accelerometer.use(tessel.port['B'])`), or used as flexible GPIO pins (e.g. `myPin = tessel.port['A'].pins[0]`).
+
+### Modules
+
+Tessel 2's module ports can be used with [Tessel modules](//tessel.io/modules) much as in [Tessel 1](http://start.tessel.io/modules).
+
+Here is an example of using the Tessel Climate module on Tessel's port B:
+
+```js
+var tessel = require('tessel');
+var climatelib = require('climate-si7020');
+var climate = climatelib.use(tessel.port['B']);
+```
+
+### Pin mapping
+
+The module ports are not just for modules! They can also be used as flexible, simply addressable GPIO pins.
+
+The pin layout for ports A and B is as follows:
+
+| Index | Function |
+|-------|----------|
+| 0     | SCL      |
+| 1     | SDA      |
+| 2     | SCK      |
+| 3     | MISO     |
+| 4     | MOSI     |
+| 5     | TX       |
+| 6     | RX       |
+| 7     | I/O      |
+
+All pins (0-7) can be used as digital pins. All pins on port B and pin 7 on port A can be used as analog pins.
+
+If you're newer to hardware and these functions look like alphabet soup to you, take a look at our [communication protocols documentation](https://tessel.io/docs/communicationProtocols) to get an idea of how these pins should be used.
+
+### Digital pins
+
+A digital pin (any pin other than 3.3V and GND on Tessel 2) is either high (on/3.3V) or low (off/0V). If unset, Tessel's pins are pulled high by default.
+
+Here is an example usage of a digital pin on Tessel:
+
+```js
+var tessel = require('tessel'); // import tessel
+var myPort = tessel.port['A']; // select the GPIO port
+var myPin = myPort.pin[2]; // select pin 2
+myPin.output(1);  // turn pin high (on)
+console.log(myPin.read()); // print the pin value to the console
+myPin.output(0);  // turn pin low (off)
+```
+
+### Analog pins
+
+An analog pin is a pin that can vary in the range between 3.3V and 0V. Pin 7 on port A and all pins on port B can read and write analog values (though pins 0 and 1 are not recommended for this purpose).
+
+Here is an example usage of an analog pin on Tessel:
+```js
+var tessel = require('tessel'); // import tessel
+var myPort = tessel.port['A']; // select the GPIO port
+var myPin = myPort.pin[7]; // select pin 2
+myPin.output(.6);  // turn pin to 60% of high
+console.log(myPin.read()); // print the pin value to the console
+```
+
+### PWM pins
+
+PWM pins are not yet implemented. See [#21](https://github.com/tessel/t2-firmware/issues/21).
+
+### SPI
+
+A SPI channel uses the SCK, MISO, and MOSI pins (2, 3, and 4 on Tessel 2). If you are unfamiliar with the SPI protocol, please see the [communication protocols tutorial](https://tessel.io/docs/communicationProtocols#spi).
+
+Here is an example using Tessel's SPI protocol:
+
+```js
+var port = tessel.port['A'];
+var spi = new port.SPI({
+  clockSpeed: 4*1000*1000, // 4MHz
+  cpol: 1, // polarity
+  cpha: 0, // clock phase
```

Comment:
This is a recommendation for the API code itself, but maybe these properties should be aliased? If they require additional comments to explain, then they aren't obvious enough, right? (I'm no stranger to making this mistake)

```js
+var spi = new port.SPI({
  clock: {
    phase: 0,
    polarity: 1,
    speed: 4000000, // 4Mhz
  }  
});
```

If you agree, I will file an issue.